### PR TITLE
use jdk21

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Use jdk 21 instead of 17

### CODE REVIEW

File: refarch-gateway/pom.xml [hunk 1]
```diff
@@ -18,7 +18,9 @@
     <name>refarch-gateway</name>
 
     <properties>
-        <java.version>17</java.version>
+        <java.version>21</java.version>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
```
Comment (devtobi) [inline refarch-gateway/pom.xml:23]: To keep this in sync with the other Java based projects I suggest using `java.version` as a variable and additionally set `maven.compiler.release`

```suggestion
        <maven.compiler.source>${java.version}</maven.compiler.source>
        <maven.compiler.target>${java.version}</maven.compiler.target>
        <maven.compiler.release>${java.version}</maven.compiler.release>
```

**Review by devtobi — CHANGES_REQUESTED** [review]
Small pom change needed, I guess you checked the rest of the dependencies (especially Spring Boot) for JDK 21 compatibility, right?

**Comment by simonhir** [comment]
> I guess you checked the rest of the dependencies (especially Spring Boot) for JDK 21 compatibility, right?

Didn't check every single dependency but for functionality and Spring-Boot and Spring-Cloud support each at least jdk 21.

**Review by devtobi — APPROVED** [review]
LGTM

**Pull request merged**